+nosnoc/+solver/MpccSolver.m

```diff
@@ -402,6 +403,37 @@
             stats.success = stats.converged;
             stats.constraint_violation = obj.compute_constraint_violation(mpcc_results.x, mpcc_results.g);
 
+            
+            if opts.calculate_stationarity_type || opts.polishing_step
+                if last_iter_failed || ~obj.complementarity_tol_met(stats) || timeout
+                    stat_type = "?";
+                    disp("Not checking stationarity due to failure of homotopy to converge");
+                else
+                    if obj.opts.homotopy_steering_strategy == HomotopySteeringStrategy.DIRECT
+                        s_elastic = [];
+                    else
+                        s_elastic = max(nlp.w.s_elastic().res);
+                    end
+                    [sol, polished_w, b_stat] = obj.check_b_stationarity(s_elastic);
+                    if ~b_stat
+                        [polished_w, res_out, stat_type, n_biactive] = obj.calculate_stationarity(true, true);
+                    else
+                        stat_type = "B";
+                    end
+                end
+                if stat_type ~= "?"
+                    mpcc_results.x = polished_w;
+                    mpcc_results.f = full(f_mpcc_fun(polished_w, nlp.p.mpcc_p().val));
+                    % TODO (@anton) also recalculate multipliers and g?
```

Comment:
the TNLP and the homotopy NLP are two different NLPs, the multipliers are completley different (except maybe in the case when the point i S stationary and the bilinear term is inactive)

Reply:
Sure but in this case I mean the multipliers on G/H which may be able to be correctly calculated in the case of S-stationarity. This would allow MpccSolver to return correct multipliers for the MPCC Lagrangian.

Reply:
Strictly speaking, if a TNLP has a solution then the multipliers are MPCC Lag. multipliers. If MPEC-SMFCQ holds they are even unique, but they dont have to be (especially for the weaker CQs) and an NLP solver will return a multiplier, depending on the internal reglarization.
If it S stat. ,then one can compute "ture" multipliers of the MPCC.